bootloader_shell.ld

```diff
@@ -64,7 +64,7 @@ SECTIONS
      */
     .app_hdr : {
         . = ALIGN(4k);
-	__app_hdr = .;
+        app_image_header = .;
 	LONG(0xdeaddead)
```

Comment:
Will these values still initialise `app_image_header`, or the 12 bytes after it?

Reply:
Not entirely sure what you mean, but I think the answer is yes (i.e. not the 12 bytes after it). This provides a symbol that can be used in code, i.e. `struct image_header app_image_header` now lives at the address indicated by `FLASH_IMGHDR` where `.app_hdr` is placed. At this location, the values `0xdeaddead`, `0x00000000` and `0xdeaddead` are found. This statement assigns the current location (`.`) but does not change it, and the size of the symbol `app_image_header` plays no role at all, so it does not move these constants. Easily verified by the way, the original `picowota.bin` before `objcopy` still has these values at the expected offset:
```
$ xxd -e picowota/picowota.bin |grep deaddead
...
00080000: deaddead 00000000 deaddead 00000000  ................
```